modyn/trainer_server/internal/grpc/trainer_server_grpc_servicer.py

```diff
@@ -160,29 +218,47 @@ def get_training_status(
         cleaned_kwargs = {k: v for k, v in response_kwargs_finished.items() if v is not None}
         return TrainingStatusResponse(**cleaned_kwargs)  # type: ignore[arg-type]
 
-    def get_final_model(
+    def store_final_model(
         self,
-        request: GetFinalModelRequest,
+        request: StoreFinalModelRequest,
         context: grpc.ServicerContext,  # pylint: disable=unused-argument
-    ) -> GetFinalModelResponse:
+    ) -> StoreFinalModelResponse:
         training_id = request.training_id
         logger.info(f"Received get final model request for training {training_id}.")
 
         if training_id not in self._training_dict:
-            logger.error(f"Training with id {training_id} has not been registered")
-            return GetFinalModelResponse(valid_state=False)
+            logger.error(f"Training with id {training_id} has not been registered.")
+            return StoreFinalModelResponse(valid_state=False)
 
         if self._training_process_dict[training_id].process_handler.is_alive():
-            logger.error(f"Training with id {training_id} is still running")
-            return GetFinalModelResponse(valid_state=False)
+            logger.error(f"Training with id {training_id} is still running.")
+            return StoreFinalModelResponse(valid_state=False)
 
         final_checkpoint_path = self._training_dict[training_id].final_checkpoint_path / "model_final.modyn"
         if final_checkpoint_path.exists():
             prefix_path = str(final_checkpoint_path.relative_to(self._modyn_base_dir))
-            return GetFinalModelResponse(valid_state=True, model_path=prefix_path)
+
+            pipeline_id = self._training_dict[training_id].pipeline_id
+            trigger_id = self._training_dict[training_id].trigger_id
+
+            register_request = RegisterModelRequest(
+                pipeline_id=pipeline_id,
+                trigger_id=trigger_id,
+                hostname=self._config["trainer_server"]["hostname"],
+                port=int(self._config["trainer_server"]["ftp_port"]),
+                model_path=prefix_path,
+            )
+
+            register_response: RegisterModelResponse = self.model_storage_stub.RegisterModel(register_request)
+
+            if not register_response.success:
+                logger.error(f"Could not store final model from training id {training_id}.")
+                return StoreFinalModelResponse(valid_state=False)
+
```

Comment:
As suggested by you, I would delete the snapshop from local disk here. This is a good idea. You are right that we clean up the temp directory anyways at the end, but if the trainer server runs for a long time, we might overflow the disk. Hence, we should clean up here

Reply:
Done in next commit.